### PR TITLE
give the ability to supply a transformErrors function to the rjsf/core

### PR DESCRIPTION
**Pull request by dagda1**
…form

Signed-off-by: Paul Cowan <paul.cowan@cutting.scot>

## Hey, I just made a Pull Request!

The `@rjsf/core` `<Form />` component has a very handy prop [transformErrors](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/form-props/#transformerrors) that gets around the problem of the default error message for the error message for a `pattern` which includes the regex and breaks security.

This allows the user to pass a `transformErrors` prop to the `NextScaffolderPage`, which gets passed through to the `next/Stepper`.

Hopefully, the test in the PR conveys the intent well.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Note the CI error:

```
*************************************************************************************
* You have uncommitted changes to the public API of a package.                      *
* To solve this, run `yarn build:api-reports` and commit all api-report.md changes. *
*************************************************************************************
```

File: plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -149,6 +151,19 @@ export const Stepper = (props: StepperProps) => {
             onSubmit={handleNext}
             fields={extensions}
             showErrorList={false}
+            transformErrors={props.transformErrors}
+            // this is needed because handleNext is really triggering the validation
+            // which only happens onSubmit
+            onChange={({ formData, schema }) => {
+              if (props.transformErrors) {
+                validator.validateFormData(
```
Comment (freben) [inline plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx:159]: Does this mean that the ENTIRE form is validated on every change, including fields lower down whose values you haven't yet had the time to enter?

**Review by freben — COMMENTED** [review]
LGTM I guess. @benjdlambert ?

File: plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx (hunk 1 above)
Comment (dagda1) [reply plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx]: @freben I meant to remove that (and I just did).  It is not needed.

**Review by freben — APPROVED** [review]
Looks good to me! Handing over to @benjdlambert

**Comment by benjdlambert** [comment]
Nice! Looks good for me!

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**